samples/update_protection_runs/update_job_run.py

```diff
@@ -0,0 +1,133 @@
+    return job.uid.id, job.uid.cluster_id, job.uid.cluster_incarnation_id
+
+def convert_to_epoch(date, end_time=False):
+    """
+    Helper function to convert date to epochtime
+    :param date:
+    :return: datetime
+    """
+    if end_time:
+        # This is to get the time till end of the day.
+        return datetime.datetime(int(date[0]), int(date[1]),
+                                 int(date[2]), 23, 59).strftime('%s')
+    return datetime.datetime(int(date[0]), int(date[1]), int(date[2])).strftime('%s')
+
+def convert_to_date(epoch):
+    """
+    Convert epoch to date
+    :param epoch:
+    :return:
+    """
+    return datetime.datetime.fromtimestamp(epoch/1000000).strftime('%c')
+
+def change_retention(job_run, job_id, cluster_id, incarnation_id, extend):
+    """
+    Change the snapshot retention time.
+    :param job_run:
+    :param job_id:
+    :param cluster_id:
+    :param incarnation_id:
+    :param extend:
+    :return:
+    """
+    body = UpdateProtectionJobRunsParam()
+    body.job_runs = []
+    index = 0
+
+    for run in job_run:
+        if run.backup_run.snapshots_deleted:
+            continue
+        body.job_runs.append(UpdateProtectionJobRun())
+        body.job_runs[index].job_uid = UniversalId()
+        body.job_runs[index].job_uid.cluster_id  = cluster_id
+        body.job_runs[index].job_uid.cluster_incarnation_id = incarnation_id
+        body.job_runs[index].job_uid.id = job_id
+        body.job_runs[index].run_start_time_usecs = run.copy_run[0].run_start_time_usecs
+        body.job_runs[index].copy_run_targets = []
+        body.job_runs[index].copy_run_targets.append(RunJobSnapshotTarget())
+        body.job_runs[index].copy_run_targets[0].days_to_keep = int(extend)
+        body.job_runs[index].copy_run_targets[0].mtype = \
+            TypeRunJobSnapshotTargetEnum.KLOCAL
+        index += 1
+    cohesity_client.protection_runs.update_protection_runs(body=body)
+
+    print ('{0:<10}\t\t{1:>24}\t\t{2:>10}\t\t{3:>10}'.format('Start Time',
+                                                             'Status',
+                                                             'Success/Error',
+                                                             'Extended '
+                                                             'Snapshot?'))
+    for run in job_run:
+        print ('{0:<10}\t\t{1:>10}\t\t{2:>10}\t\t{3:>10}'.format(
+            str(convert_to_date(run.backup_run.stats.start_time_usecs)),
+            run.backup_run.status,
+            str(job_run[0].backup_run.stats.num_successful_tasks) + '/' +
+            str(job_run[0].backup_run.stats.num_failed_tasks), 'Yes'))
+
+def main(args):
+
+    start_time = convert_to_epoch(args.start_time.split('-')) + MILLISECONDS
+    end_time = convert_to_epoch(args.end_time.split('-'), True) + MILLISECONDS
+
+    job_id, cluster_id, incarnation_id = get_pro_id(args.job_name)
```

Comment:
Exception handling for get_pro_id empty response

Reply:
handled in the function.